src/guess_density.jl

```diff
@@ -14,46 +17,45 @@ function random_density(basis::PlaneWaveBasis; n_electrons=basis.model.n_electro
     ρ_from_total_and_spin(ρtot, ρspin)
 end
 
-
 @doc raw"""
-    guess_density(basis, magnetic_moments=[])
```

Comment:
This makes sense; I changed it just because I was adding the `method` kwarg and I wasn't sure how clear 
```julia
guess_density(basis, magnetic_moments=[]; method=:smart)
```
would be. Moving to actual multiple-dispatch based on `GuessDensityMethod` subtypes, i.e. 
```julia
guess_density(::GuessDensityMethod, basis, magnetic_moments=[])
```
makes for a cleaner method signature IMO.

Reply:
I generally agree with this idea. The remaining question for me is where to put the algorithm argument. Following the common solve interface in Julia the tendency seems to be to put the algorithm after the problem description, so along this idea I would have:

```
guess_density(basis, magnetic_moments, ::GuessDensityMethod)
```
and a 
```
guess_density(basis, m::AbstractGuessDensityMethod) = guess_density(basis, [], m)
```
(where the names are to be discussed).

Reply:
I think the generic solution to the "what order should optional arguments be" is : kwargs. Optional positional arguments with default values are tricky, kwargs are much nicer. Esp with the new f(; x) syntax, there's really not much of a reason to use optional positional arguments.